src/lib/swapper/swappers/CowSwapper/cowGetTradeTxs/cowGetTradeTxs.ts

```diff
@@ -14,44 +13,30 @@ export async function cowGetTradeTxs(
   deps: CowSwapperDeps,
   input: TradeResult,
 ): Promise<Result<TradeTxs, SwapErrorRight>> {
-  try {
-    const getOrdersResponse = await cowService.get<CowSwapGetOrdersResponse>(
-      `${deps.apiUrl}/v1/orders/${input.tradeId}`,
-    )
+  const maybeGetOrdersResponse = await cowService.get<CowSwapGetOrdersResponse>(
+    `${deps.apiUrl}/v1/orders/${input.tradeId}`,
+  )
 
-    const {
-      data: { status },
-    } = getOrdersResponse
+  if (maybeGetOrdersResponse.isErr()) return Err(maybeGetOrdersResponse.unwrapErr())
 
-    if (status !== ORDER_STATUS_FULFILLED) {
-      return Ok({
-        sellTxid: '',
-      })
-    }
-
-    const getTradesResponse = await cowService.get<CowSwapGetTradesResponse>(
-      `${deps.apiUrl}/v1/trades/?orderUid=${input.tradeId}`,
-    )
+  const {
+    data: { status },
+  } = maybeGetOrdersResponse.unwrap()
 
+  if (status !== ORDER_STATUS_FULFILLED) {
     return Ok({
-      sellTxid: input.tradeId,
-      buyTxid: getTradesResponse.data[0].txHash,
+      sellTxid: '',
     })
-  } catch (e) {
-    if (e instanceof SwapError)
-      return Err(
-        makeSwapErrorRight({
-          message: e.message,
-          code: e.code,
-          details: e.details,
-        }),
-      )
-    return Err(
-      makeSwapErrorRight({
-        message: '[cowGetTradeTxs]',
-        cause: e,
-        code: SwapErrorType.GET_TRADE_TXS_FAILED,
-      }),
-    )
   }
+
+  const maybeGetTradesResponse = await cowService.get<CowSwapGetTradesResponse>(
+    `${deps.apiUrl}/v1/trades/?orderUid=${input.tradeId}`,
+  )
+
+  return maybeGetTradesResponse.map(getTradesResponse => {
```

Comment:
Unchecked monad.

Reply:
Fixed as part of https://github.com/shapeshift/web/commit/a45340d5415b9d8084ee8bb757678f5a9dfad3db